aten/src/ATen/native/cpu/SparseFactories.cpp

```diff
@@ -66,9 +66,30 @@ void _spdiags_kernel_cpu(
       });
 }
 
+void _spdiags_backward_kernel_cpu(TensorIterator& iter, Tensor& grad_in) {
+  AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND4(
+      at::ScalarType::BFloat16,
+      at::ScalarType::Half,
+      at::ScalarType::Bool,
+      at::ScalarType::ComplexHalf,
+      iter.dtype(),
+      "spdiags_backward_cpu",
+      [&] {
+        auto grad_in_accessor = grad_in.accessor<scalar_t, 2>();
+        cpu_kernel(
+            iter,
+            [&](scalar_t grad_out_value,
+                int64_t row_idx,
+                int64_t col_idx) -> scalar_t {
+              grad_in_accessor[row_idx][col_idx] = grad_out_value;
```

Comment:
@pearu changing this line to `grad_in_accessor[row_idx][col_idx] += grad_out_value` should correctly handle the uncoalesced `grad_out` case?